.github/workflows/unit-and-integration-tests.yaml

```diff
@@ -0,0 +1,169 @@
+    name: Integration tests on Testflinger instance with Intel GPU
+    runs-on: [self-hosted, testflinger]
+    steps:
+      - name: Submit job to testflinger
+        uses: canonical/testflinger/.github/actions/submit@main
+        with:
+          poll: true
+          job: |
+            job_queue: dell-precision-3470-c30322
+            provision_data:
+              distro: jammy
+              user_data: |
+                #cloud-config
+                # Note, much of this is borrowed from the
+                # charm-dev multipass template:
+                # https://github.com/canonical/multipass-blueprints/blob/main/v1/charm-dev.yaml
+                package_update: true
+                package_upgrade: true
+                package_reboot_if_required: true
+
+                packages:
+                - python3-pip
+
+                snap:
+                  commands:
+                  # Juju 3.5 is supported until Jan 2025
+                  # https://juju.is/docs/juju/roadmap
+                  - snap install juju --channel=3.5/stable
+                  - snap install microk8s --channel=1.29-strict/stable
+                  - snap install --classic charmcraft
+                  - snap refresh
+
+                runcmd:
+                - |
+                  # Clone source from the appropriate branch/commit
+                  sudo -u ubuntu git clone \
+                    -b ${{ github.head_ref || github.ref_name }} \
+                    https://github.com/canonical/intel-device-plugins-k8s-operator.git \
+                    ~ubuntu/intel-device-plugins-k8s-operator
+
+                - |
+                  # Install tox from pypi (v4) instead of apt (v3)
+                  pip3 install tox
+
+                - |
+                  # Make sure juju directory is there
+                  # https://bugs.launchpad.net/juju/+bug/1995697
+                  sudo -u ubuntu mkdir -p /home/ubuntu/.local/share/juju
+
+                - |
+                  # setup lxd for charmcraft
+                  lxd init --auto
+                  adduser ubuntu lxd
+
+                - |
+                  # setup microk8s and bootstrap
+                  usermod -a -G snap_microk8s ubuntu
+                  usermod -a -G microk8s ubuntu
+                  microk8s status --wait-ready
+                  # wait for storage to become available
+                  microk8s.enable hostpath-storage
+                  microk8s.kubectl rollout status \
+                    deployments/hostpath-provisioner \
+                    -n kube-system -w --timeout=600s
+                  sudo -u ubuntu juju bootstrap microk8s microk8s
+                  sudo -u ubuntu juju add-model \
+                    --config logging-config="<root>=WARNING; unit=DEBUG" \
+                    --config update-status-hook-interval="1m" \
+                    welcome-k8s
+            test_data:
+              test_cmds: |
+
+                # Exit immediately if a test fails
```

Comment:
Can we consider putting these commands also in the script in the repository itself ?

Reply:
Yes, good idea. I'll give it a try.

Reply:
I created a small script for testing arbitrary commands wrapped in a while loop (to give cloud-init time to install all the dependencies), making the `test_cmds` section a lot cleaner.